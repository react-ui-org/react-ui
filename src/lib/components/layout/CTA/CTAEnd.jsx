import PropTypes from 'prop-types';
import React from 'react';
import styles from './CenteredCTA.scss';

const CenteredCTAEnd = (props) => {
  const {
    children,
  } = props;

  return (
    <div className={styles.end}>
      {children}
    </div>
  );
};

CenteredCTAEnd.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default CenteredCTAEnd;
