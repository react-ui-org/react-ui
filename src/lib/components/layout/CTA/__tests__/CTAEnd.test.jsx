import React from 'react';
import {
  render,
  within,
} from '@testing-library/react';
import { CTAEnd } from '../CTAEnd';

describe('rendering', () => {
  it.each([
    [
      { children: <div>content text</div> },
      (rootElement) => expect(within(rootElement).getByText('content text')),
    ],
  ])('renders with props: "%s"', (testedProps, assert) => {
    const dom = render((
      <CTAEnd
        {...testedProps}
      />
    ));

    assert(dom.container.firstChild);
  });
});
