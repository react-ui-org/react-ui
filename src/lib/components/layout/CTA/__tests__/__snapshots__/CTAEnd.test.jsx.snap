// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with a single child 1`] = `
<div
  className="end"
>
  <span>
    content
  </span>
</div>
`;

exports[`rendering renders correctly with multiple children 1`] = `
<div
  className="end"
>
  <span>
    content 1
  </span>
  <span>
    content 2
  </span>
  <span>
    content 3
  </span>
</div>
`;
